lib/Dialect/Utils.h

```diff
@@ -35,6 +40,122 @@ FailureOr<int64_t> get1DExtractionIndex(Op op) {
   return insertOffsetAttr.getInt();
 }
 
+// Refactored out verifiers and inferReturnTypes for BGV and CKKS operations.
+template <typename Op>
+LogicalResult verifyMulOp(Op* op) {
```

Comment:
nit: I would maybe label these as RLWE specific (e.g. verifyRlweMulOp) or move this to a file specific for RLWE verifiers